src/codewhisperer/service/transformByQHandler.ts

```diff
@@ -114,18 +114,36 @@ export async function validateProjectSelection(project: vscode.QuickPickItem) {
     const baseCommand = 'javap'
     const args = ['-v', classFilePath]
     const spawnResult = spawnSync(baseCommand, args, { shell: false, encoding: 'utf-8' })
-
-    if (spawnResult.error || spawnResult.status !== 0) {
+    if (spawnResult.status !== 0) {
         void vscode.window.showErrorMessage(CodeWhispererConstants.noSupportedJavaProjectsFoundMessage)
+        let errorLog = ''
+        errorLog += spawnResult.error ? `${JSON.stringify(spawnResult.error)}` : ''
+        errorLog += `${spawnResult.stderr}\n${spawnResult.stdout}`
+        getLogger().error(`CodeTransform: Error in running javap command = ${errorLog}`)
+        let errorReason = ''
+        if (spawnResult.stdout) {
+            errorReason = 'JavapExecutionError'
+            // should never happen -- stdout from javap has always been much, much smaller than the default buffer limit of 1MB
+            if (Buffer.byteLength(spawnResult.stdout, 'utf-8') > CodeWhispererConstants.maxBufferSize) {
+                errorReason += '-BufferOverflow'
+            }
+        } else {
+            errorReason = 'JavapSpawnError'
+        }
+        if (spawnResult.error) {
+            // oddly, the 'code' field is not visible until I stringify the object, but then I have to parse it to access 'code'
+            // 'code' is a high-level symbol representing the error (ex. 'ENOENT', 'ENOBUFS', etc.)
+            const errorCode = JSON.parse(JSON.stringify(spawnResult.error)).code ?? 'UNKNOWN'
```

Comment:
Did you try `as any` ? `parse(stringify(...))` is a pretty heavy workaround.

```suggestion
            const errorCode = (spawnResult.error as any).code ?? 'UNKNOWN'
```

Though whatever problem you ran into should go away if you switch to using ChildProcess https://github.com/aws/aws-toolkit-vscode/blob/961ff52e05ed45d25c749a2237de6b2b3237f076/src/shared/utilities/childProcess.ts#L68

Reply:
Thanks. `as any` worked.